common-practices-tools/contribution/contrib-first.md

```diff
@@ -0,0 +1,46 @@
+# Contrib First
+
+Whenever we are building something that could be of use to more than one project or client, we build it as contributed work first.
+
+## Rationale for contrib first
+
+-   **Fiscal responsibility** - Building it and contributing it means that other government agencies will never have to pay to build the same thing. This helps agencies comply with Federal Source Code Policy: Achieving Efficiency, Transparency, and Innovation through Reusable and Open Source Software [OMB Memorandum M-16-21](https://obamawhitehouse.archives.gov/sites/default/files/omb/memoranda/2016/m_16_21.pdf)
+-   **Reusablity** - CivcActions other clients can benefit from work that was already done.
+-   **Security** - contributing our work to an open source project like Drupal means it may receive security coverage by the Drupal security team and the public.
+-   **Avoiding the gift that never happens** - Clients are not typically supportive of taking working local software that was already built for them and in use by them and then paying to move or refactor that software to be open source. The benefit is too small for the cost. By building it as contributed code first, there is no extra cost.
+-   **Development happens in the open** - The issue are public. The commits are public. Everyone can contribute.
+
+## Contrib first patches
```

Comment:
This is pretty Drupal specific language - it would be nice to write this in more general terms (using Drupal as an example, or translating terms etc) as contribution to all the many FOSS projects we use should be our goal.

Reply:
Good point.  It does make a fare amount of assumptions related to Drupal.  I wanted this to be the reference specifically for Drupal that I could point to, mainly because the whole issue queue and contrib credit does not really exist elsewhere.  What if doubled down on making this one section Drupal specific and then added another section that was more agnostic for other FOSS contributions?

Reply:
I think that would work!